site/source/docs/porting/pthreads.rst

```diff
@@ -43,17 +43,67 @@ but is unrelated. That flag does not use pthreads or SharedArrayBuffer, and
 instead uses a plain Web Worker to run your main program (and postMessage to
 proxy messages back and forth).
 
+Blocking on the main browser thread
+===================================
+
+The Web API for atomics does not allow blocking on the main thread
+(specifically, ``Atomics.wait`` doesn't work there). Such blocking is
+necessary in APIs like ``pthread_join`` and anything that uses a futex wait
+under the hood, like ``usleep()``, ``emscripten_futex_wait()``, or
+``pthread_mutex_lock()``. To make them work, we use a busy-wait on the main
+browser thread, which can make the browser tab unresponsive, and also wastes
+power. (On a pthread, this isn't a problem as it runs in a Web Worker, where
+we don't need to busy-wait.)
+
+Busy-waiting on the main browser thread in general will work despite the
+downsides just mentioned, for things like waiting on a mutex.
+However, things like ``pthread_join`` and ``pthread_cond_wait``
+can block for long periods of time, and if that
```

Comment:
Technically of course mutex locking can block a long time too. Here I might say something like "calls to `pthread_join` and `pthread_cond_wait` are often intended to block for long periods of time" just to emphasize the difference.